quiet-spans/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+use tracing::Span;
+use tracing::Subscriber;
+use tracing_subscriber::filter::dynamic_filter_fn;
+use tracing_subscriber::filter::Directive;
+use tracing_subscriber::registry::LookupSpan;
+use tracing_subscriber::Layer;
+
+const TARGET: &str = "quiet_children";
+const ALWAYS_QUIET: &str = "always_quiet_children";
+const SOMETIMES_QUIET: &str = "sometimes_quiet_children";
+
+/// Children of this span are always ignored.
+pub fn always_quiet_children() -> Span {
+    tracing::info_span!(target: TARGET, ALWAYS_QUIET)
+}
+
+/// Children of this span are ignored if `verbose_spans` is false in [`disable_noisy_spans`]
+pub fn sometimes_quiet_children() -> Span {
+    tracing::info_span!(target: TARGET, SOMETIMES_QUIET)
+}
+
+/// Layer which disables noisy spans - children of [`always_quiet_children`] and children of
+/// [`sometimes_quiet_children`] (when `verbose_spans` is false).
+pub fn disable_noisy_spans<S>(verbose_spans: bool) -> impl Layer<S>
```

Comment:
nit: it's hard to read such API at callsites if it's a bool. Rust has expresive syntax, an enum `NoisySpanLevel  { Some, All}` or something similar would go a step further towards a readable API. 
(then some of the comment can go there)

`disable_noisy_spans(false)` might be interpreted as *not* disabling them aftor all!😝

Reply:
I'll open a follow-up PR to clean this up, in the meantime I think it's worth merging.

Reply:
Thanks :)